tests/zentorch/testing_utils.py

```diff
@@ -0,0 +1,171 @@
+    elif task == "audio-classification":
+        return AutoModelForAudioClassification.from_pretrained(model_id)
+    elif task == "image-to-text":
+        return AutoModelForVision2Seq.from_pretrained(model_id)
+    elif task == "visual-question-answering":
+        return AutoModelForVisualQuestionAnswering.from_pretrained(model_id)
+    else:
+        raise ValueError(f"Task {task} not supported")
+
+
+def load_diffusion_pipeline(pipeline_id: str, task: str):
+    if task == "text-to-image":
+        return DiffusionPipeline.from_pretrained(pipeline_id)
+    else:
+        raise ValueError(f"Task {task} not supported")
+
+
+def get_transformers_model_inputs(model_id: str, task: str):
+    if task in ["text-classification", "text-generation", "text2text-generation"]:
+        processor = AutoTokenizer.from_pretrained(model_id)
+        text = ["This is a test sentence"] * BATCH_SIZE
+        dummy_inputs = processor(text=text, return_tensors="pt")
+
+    elif task in ["image-to-text", "image-classification"]:
+        processor = AutoImageProcessor.from_pretrained(model_id)
+        image_url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        images = [PIL.Image.open(requests.get(image_url, stream=True).raw)] * BATCH_SIZE
+        dummy_inputs = processor(images=images, return_tensors="pt")
+
+    elif task in ["audio-classification", "automatic-speech-recognition"]:
+        processor = AutoFeatureExtractor.from_pretrained(model_id)
+        audio_url = "https://huggingface.co/datasets/Narsil/asr_dummy/resolve/main/mlk.flac"
+        audios = [ffmpeg_read(requests.get(audio_url).content, processor.sampling_rate)] * BATCH_SIZE
+        dummy_inputs = processor(raw_speech=audios, return_tensors="pt")
+
+    elif task == "visual-question-answering":
+        processor = AutoProcessor.from_pretrained(model_id)
+        text = ["What is in the image?"] * BATCH_SIZE
+        image_url = "http://images.cocodataset.org/val2017/000000039769.jpg"
+        images = [PIL.Image.open(requests.get(image_url, stream=True).raw)] * BATCH_SIZE
+        dummy_inputs = processor(text=text, images=images, return_tensors="pt")
+
+    else:
+        raise ValueError(f"Task {task} not supported")
+
+    return dummy_inputs
+
+
+def get_diffusion_pipeline_inputs(pipeline_id: str, task: str):
+    dummy_inputs = {}
+
+    if task == "text-to-image":
+        dummy_inputs["prompt"] = ["This is a test prompt"] * BATCH_SIZE
+    else:
+        raise ValueError(f"Task {task} not supported")
+
+    return dummy_inputs
+
+
+def compile_transformers_model(model, backend):
+    torch._dynamo.reset()
+
+    model = torch.compile(model, backend=backend)
+
+    return model
+
+
+def compile_diffusion_pipeline(pipeline, backend):
+    torch._dynamo.reset()
+
+    pipeline.unet = torch.compile(pipeline.unet, backend=backend)
+    pipeline.vae.decoder = torch.compile(pipeline.vae.decoder, backend=backend)
+
+    return pipeline
```

Comment:
I think it would be nice to add tests with `fullgraph=True` as well, but that can be done later